File: PartyTime/src/main/java/com/pasciak/partytime/entities/EventInvite.java
package com.pasciak.partytime.entities;

import java.util.Objects;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonIncludeProperties;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name="event_invite")
public class EventInvite {
	
	@Id
	@Column(name="id")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(name="comment")
	private String comment;
	
	@Column(name="attending")
	private Integer attending;
	
	@ManyToOne
	@JoinColumn(name = "user_id")
	private User user;
	
	@ManyToOne
	@JoinColumn(name = "event_id")
	private Event event;
	
	EventInvite() {
		super();
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Integer getAttending() {
		return attending;
	}

	public String getComment() {
		return comment;
	}

	public void setComment(String comment) {
		this.comment = comment;
	}

	public void setAttending(Integer attending) {
		this.attending = attending;
	}
	
	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Event getEvent() {
		return event;
	}

	public void setEvent(Event event) {
		this.event = event;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		EventInvite other = (EventInvite) obj;
		return Objects.equals(id, other.id);
	}

	@Override
	public String toString() {
		return "EventInvite [id=" + id + ", comment=" + comment + ", attending=" + attending + "]";
	}
	
}

File: ngPartyTime/src/app/models/event-invite.ts
import { User } from './user';
import { Event } from './event';

export class EventInvite {
  id: number;
  userId: number;
  eventId: number;
  comment?: string | null;
  attending?: number | null;
  user?: User | null;
  event?: Event | null;

  constructor(
    id: number = 0,
    userId: number = 0,
    eventId: number = 0,
    comment?: string | null,
    attending?: number | null,
    user: User | null = null,
    event: Event | null = null
  ) {
    this.id = id ?? 0;
    this.userId = userId ?? 0;
    this.eventId = eventId ?? 0;
    this.comment = comment ?? null;
    this.attending = attending ?? null;
    this.user = user ?? null;
    this.event = event ?? null;
  }
}
